bayesian/utility/user_utils.py

```diff
@@ -54,6 +58,43 @@ def create_or_update_user(user_id, snyk_api_token, user_source):
         raise UserException("Error updating user") from e
 
 
+def create_or_update_user_in_cache(user_id, snyk_api_token, user_source):
+    """Create or Update User in cache."""
+    if ENABLE_USER_CACHING:
+        db_cache_file_path = DB_CACHE_DIR + "/" + user_id + ".json"
+        user_detail = {"user_id": user_id,
+                       "user_source": user_source,
+                       "snyk_api_token": snyk_api_token,
+                       "status": UserStatus.REGISTERED.name,
+                       "created_date": datetime.datetime.now(),
+                       "registered_date": datetime.datetime.now()}
+        try:
+            with open(db_cache_file_path, 'w', encoding='utf-8') as file:
+                json.dump(user_detail, file, ensure_ascii=False, indent=4, default=str)
+
+            logger.info("User added in cache with id %s", user_id)
+        except Exception as e:
+            logger.exception("Error while caching user with id %s", user_id)
+            raise UserException("Error caching user") from e
+
+
+def get_user_from_cache(user_id):
+    """Get User from cache."""
+    user = {}
+    try:
+        db_cache_file_path = DB_CACHE_DIR + "/" + user_id + ".json"
+
+        if os.path.isfile(db_cache_file_path):
+            logger.info("Found user in cache with id %s", user_id)
+            file = open(db_cache_file_path, "r")
+            user = json.loads(file.read())
+        return user
+    except Exception as e:
+        logger.info("User not found in cache with id %s", user_id)
+        logger.error(e)
```

Comment:
For unregistered users condition wont satisfy and it will return {}, error will be printed only in case of any exception while reading.
https://github.com/jparsai/fabric8-analytics-server/blob/11e979ef27a31a1193176c68da6ed4d47624bbc9/bayesian/utility/user_utils.py#L87

Reply:
Ack